src/server/actions/Reporting.ts

```diff
@@ -0,0 +1,75 @@
+import { EventVolunteerResponse } from '@/types/dataModel/eventVolunteer';
+import { getEvent } from './Event';
+import { EventResponse } from '@/types/dataModel/event';
+import { getEventVolunteersByOrganization } from './EventVolunteers';
+import { getRangesOverlap } from '@/utils/math';
+
+/**
+ * Gets statistics for volunteering efforts for an organization, optionally bounded to events within a search range
+ * @param organizationId Id of organization to generate statistics for
+ * @param startDate Lower bound on search range for event volunteer records
+ * @param endDate Upper bound on search range for event volunteer records
+ * @returns Total volunteer hours and number of unique volunteers associated with the organization within the search range
+ */
+export async function getOrganizationReport(
+  organizationId: string,
+  startDate?: Date,
+  endDate?: Date
+): Promise<{
+  organization: string;
+  num_volunteers: number;
+  num_hours: number;
+}> {
+  // Return object
+  const report = {
+    organization: organizationId,
+    num_volunteers: 0,
+    num_hours: 0,
+  };
+
+  // Get search range
+  const fromTime = startDate?.valueOf();
+  const toTime = endDate?.valueOf();
+
+  // If valid search range, calculate num volunteers and hours
+  if (fromTime == undefined || toTime == undefined || toTime > fromTime) {
+    // Get all event volunteers under organization
+    const eventVolunteers: EventVolunteerResponse[] =
+      await getEventVolunteersByOrganization(organizationId);
+
+    // Iterate over eventVolunteers to get unique volunteer ids and total time for all records within search range
+    const volunteerIds: Set<string> = new Set<string>();
+    let volunteerTime: number = 0;
+    for (const eventVolunteer of eventVolunteers) {
+      const event: EventResponse = await getEvent(eventVolunteer.event);
+
+      // Get event time range
+      let timeRange: [number, number] | undefined = [
```

Comment:
i dont think this type is right, no need to explicitly set it here

Reply:
the reason I included `| undefined` is because a couple lines down it gets set to the result of `getRangesOverlap()` which is `[number, number] | undefined`, because if there is no overlap between the ranges in the func arg it returns undefined, a behavior I use to make sure I don't count volunteers who volunteered w the org at events outside the query range but none within the query range